csharp/src/Ice/Connection.cs

```diff
@@ -50,39 +52,20 @@ public ConnectionInfo(ConnectionInfo underlying, bool incoming, string adapterNa
 
     public delegate void CloseCallback(Connection con);
 
-    /// <summary>
-    /// Determines the behavior when manually closing a connection.
-    /// </summary>
-    public enum ConnectionClose
+    public interface Connection
     {
         /// <summary>
-        /// Close the connection immediately without sending a close connection protocol message to the peer and waiting
-        ///  for the peer to acknowledge it.
+        /// Closes the connection forcefully without waiting for any other operation to complete.
         /// </summary>
+        void abort();
 
-        Forcefully,
         /// <summary>
-        /// Close the connection by notifying the peer but do not wait for pending outgoing invocations to complete.
-        /// On the
-        ///  server side, the connection will not be closed until all incoming invocations have completed.
+        /// Closes the connection gracefully.
         /// </summary>
-
-        Gracefully,
-        /// <summary>
-        /// Wait for all pending invocations to complete before closing the connection.
-        /// </summary>
-
-        GracefullyWithWait
-    }
-
-    public interface Connection
-    {
-        /// <summary>
-        /// Manually close the connection using the specified closure mode.
-        /// </summary>
-        /// <param name="mode">Determines how the connection will be closed.
-        ///  </param>
-        void close(ConnectionClose mode);
+        /// <param name="waitForInvocations">If true, this operation waits for all pending invocations to complete
+        /// before closing the connection.</param>
+        /// <returns>A task that completes when the connection is closed.</returns>
+        Task closeAsync(bool waitForInvocations = true);
```

Comment:
I guess we're sticking with camelCase for consistency with the rest of the API?

Reply:
Yes.